Android/app/src/go/intra/split/retrier.go

```diff
@@ -318,14 +318,21 @@ func (r *retrier) Write(b []byte) (int, error) {
 			if err == nil {
 				return n, nil
 			}
+
+			leftover := b[n:]
 			// A write error occurred on the provisional socket.  This should be handled
 			// by the retry procedure.  Block until we have a final socket (which will
 			// already have replayed b[:n]), and retry.
 			<-r.retryCompleteFlag
+	
 			r.mutex.Lock()
 			r.mutex.Unlock()
-			m, err := r.conn.Write(b[n:])
-			return n + m, err
+	
```

Comment:
Please also add a comment here to explain the reason we are skipping 0-len buffers.